package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -242,60 +245,86 @@ def read_DATA_timestep(self, n_atoms, TS_class, TS_kwargs):
 
         unitcell = self._parse_box(header)
 
-        positions = np.zeros((n_atoms, 3),
-                             dtype=np.float32, order='F')
         try:
-            self._parse_pos(sects['Atoms'], positions)
+            positions, ordering = self._parse_pos(sects['Atoms'])
         except KeyError:
             raise IOError("Position information not found")
 
         if 'Velocities' in sects:
-            velocities = np.zeros((n_atoms, 3),
-                                  dtype=np.float32, order='F')
-            self._parse_vel(sects['Velocities'], velocities)
+            velocities = self._parse_vel(sects['Velocities'], ordering)
         else:
             velocities = None
 
         ts = TS_class.from_coordinates(positions,
                                        velocities=velocities,
                                        **TS_kwargs)
-        ts._unitcell = unitcell
+        ts.dimensions = unitcell
```

Comment:
I am looking at this line out of context but are you sure that `unitcell` is in the canonical format and can be assigned to `ts.dimensions`?  Isn't `unitcell` the native representation that should be stored in `ts._unitcell` while assigning to `ts.dimensions` performs a back transformation to native format?